server/mailet/integration-testing/src/test/java/org/apache/james/mailets/RemoteDeliveryDKIMIntegrationTest.java

```diff
@@ -0,0 +1,279 @@
+        "selector", "example.com");
+
+    @RegisterExtension
+    static MockSmtpServerExtension mockSmtpServerExtension = new MockSmtpServerExtension();
+    @TempDir
+    static File tempDir;
+
+    @RegisterExtension
+    SMTPMessageSenderExtension smtpSenderExtension = new SMTPMessageSenderExtension(Domain.of(DEFAULT_DOMAIN));
+
+    private TemporaryJamesServer jamesServer;
+    private DataProbe dataProbe;
+    private DKIMVerifier dkimVerifier;
+
+    @BeforeEach
+    void setUp() {
+        dkimVerifier = new DKIMVerifier(MOCK_PUBLIC_KEY_RECORD_RETRIEVER);
+    }
+
+    @AfterEach
+    void tearDown() {
+        if (jamesServer != null) {
+            jamesServer.shutdown();
+        }
+    }
+
+    @Nested
+    class WhenEnable8BitMime {
+        @Disabled("JAMES-3016 assertion failed:" +
+            "org.apache.james.jdkim.exceptions.PermFailException: Computed bodyhash is different from the expected one")
+        @Test
+        void remoteDeliveryShouldNotBreakDKIMSignWhen7BitTextMessage(SMTPMessageSender messageSender, DockerMockSmtp dockerMockSmtp) throws Exception {
+
+            InMemoryDNSService inMemoryDNSService = new InMemoryDNSService()
+                .registerMxRecord(JAMES_ANOTHER_DOMAIN, dockerMockSmtp.getIPAddress());
+
+            jamesServer = TemporaryJamesServer.builder()
+                .withBase(SMTP_ONLY_MODULE)
+                .withOverrides(binder -> binder.bind(DNSService.class).toInstance(inMemoryDNSService))
+                .withMailetContainer(TemporaryJamesServer.SIMPLE_MAILET_CONTAINER_CONFIGURATION
+                    .putProcessor(directResolutionTransport(MailetConfiguration.remoteDeliveryBuilder()
+                        .addProperty("mail.smtp.allow8bitmime", "true")))
+                    .putProcessor(CommonProcessors.bounces()))
+                .build(tempDir);
+
+            dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+            dataProbe.addDomain(DEFAULT_DOMAIN);
+            dataProbe.addUser(FROM, PASSWORD);
+
+            FakeMail mail = FakeMail.builder()
+                .name("a-mail-with-7bit-encoding")
+                .sender(new MailAddress(FROM))
+                .recipient(new MailAddress(RECIPIENT))
+                .mimeMessage(MimeMessageUtil.mimeMessageFromStream(
+                    ClassLoader.getSystemResourceAsStream("eml/message-text-only-7bit.eml")))
+                .build();
+            messageSender.connect(LOCALHOST_IP, jamesServer.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .sendMessage(mail);
+
+            MimeMessage sendMessage = toMimeMessage(getFirstRecivedMail(dockerMockSmtp));
+
+            assertThat(dkimVerifier.verifyUsingCRLF(sendMessage))
+                .isNotEmpty();
+        }
+
+        @CsvSource({
+            "a-mail-with-7bit-base64-encoding, eml/message-multipart-7bit.eml",
+            "a-mail-with-8bit-encoding, eml/message-text-only-8bit.eml",
```

Comment:
FYI the reason 8bit email doesn't break the test while it's supposed to be. That's this issue https://issues.apache.org/jira/browse/JAMES-1824

8 bit lines are altered while retrieving them in `LineHandlerUpstreamHandler` for example. It's up to Netty.

`A 8bit encoded body with €uro symbol.` will be `A 8bit encoded body with ?uro symbol.` on James side

Reply:
see #3068

Reply:
Update, the `SMTPMessageSender` cannot send the mail containing Unicode characters correctly